addons/event/static/src/client_action/event_registration_summary_dialog.js

```diff
@@ -73,6 +74,25 @@ export class EventRegistrationSummaryDialog extends Component {
     async onRegistrationConfirm() {
         await this.orm.call("event.registration", "action_set_done", [this.registration.id]);
         this.registrationStatus.value = "confirmed_registration";
+        this.props.close();
+        if (this.props.model) {
+            this.props.model.load();
+        }
+        if (this.props.doNextScan) {
+            this.onScanNext();
+        }
+    }
+
+    async undoRegistration() {
+        if (["confirmed_registration", "already_registered"].includes(this.registrationStatus.value)) {
+            await this.orm.call("event.registration", "action_confirm", [this.registration.id]);
+        } else if (this.registrationStatus.value == "uconfirmed_registration") {
```

Comment:
Seems it was not completely tested :disguised_face: 
```suggestion
        } else if (this.registrationStatus.value == "unconfirmed_registration") {
```